client/src/partials/reference_group/reference_group.js

```diff
@@ -0,0 +1,126 @@
+          'reference_group' : {
+            columns : ['id', 'is_report', 'reference_group', 'text', 'position', 'section_bilan_id']
+          },
+          'section_bilan' : {
+            columns : ['text::section_bilan_txt']
+          }          
+        },
+        join : [
+          'reference_group.section_bilan_id=section_bilan.id'
+        ],
+      }
+    };
+
+    dependencies.section_bilan = {
+      query : {
+        tables : {
+          'section_bilan' : {
+            columns : ['id', 'text']
+          }
+        }
+      }
+    };
+
+    dependencies.enterprise = {
+      query : {
+        tables : {
+          'enterprise' : {
+            columns : ['currency_id']
+          },
+          'currency' : {
+            columns : ['id', 'symbol']
+          }
+        },
+        join : [
+            'enterprise.currency_id=currency.id'
+          ]
+        }
+    };
+
+    function startup (models) {
+      angular.extend($scope, models);
+    }
+
+    appstate.register('enterprise', function (enterprise) {
+      $scope.enterprise = enterprise;
+      validate.process(dependencies)
+      .then(startup);
+    });
+
+    $scope.delete = function (reference_group) {
+      var result = confirm($translate.instant('REFERENCE_GROUP.CONFIRM'));
+      if (result) {  
+        connect.basicDelete('reference_group', reference_group.id, 'id')
+        .then(function () {
+          $scope.reference_groups.remove(reference_group.id);
+          messenger.info($translate.instant('REFERENCE_GROUP.DELETE_SUCCESS'));
+        });
+      }
+    };
+
+    $scope.edit = function (reference_group) {
+      session.action = 'edit';
+      session.edit = angular.copy(reference_group);
+      session.edit.is_report = session.edit.is_report !== 0;
+    };
+
+    $scope.new = function () {
+      session.action = 'new';
+      session.new = {};
+    };
+
+    $scope.save = {};
+
+    $scope.save.edit = function () {
+      var record = connect.clean(session.edit);
+      delete record.reference;
+      delete record.section_bilan_txt;
+      record.is_report = (session.edit.is_report)?1:0;
+
+      connect.basicPost('reference_group', [record], ['id'])
+      .then(function () {
+        messenger.success($translate.instant('REFERENCE_GROUP.UPDATE_SUCCES')); 
+        $scope.reference_groups.put(record);
+        session.action = '';
+        session.edit = {};
+      });
+    };
+
+    $scope.save.new = function () {
+      var record = connect.clean(session.new);
+      record.is_report = (session.new.is_report)?1:0;
+
+      connect.basicPut('reference_group', [record])
+      .then(function () {
+        messenger.success($translate.instant('REFERENCE_GROUP.SAVE_SUCCES'));        
+        record.reference = generateReference(); 
+        $scope.reference_groups.post(record);
+        session.action = '';
+        session.new = {};
+      });
+    };
+
+    function generateReference () {
+      window.data = $scope.reference_groups.data;
```

Comment:
Why are you attaching things to the `window` object?  If you need to use the browser window, you should import angular's $window service.
```js
angular.controller('someController', ['$window', function ($window) {
  // use '$window' where you would use 'window'
  // ...
});
```
It is safer.  See the documentation [here](https://docs.angularjs.org/api/ng/service/$window).